### ISSUE TITLE
warp.deploy method not working with Arweave.app wallet

### ISSUE THREAD

**Issue by luckyr13** [warp-contracts/warp-contracts-plugins#15]
Hello friends, how is it going?

Today I updated the libs from one of my projects. I updated `warp-contracts` from `1.2.53` to `1.3.3`. I'm pretty happy with all the improvements but I noticed that contracts deployment are not working when using `Arweave.app` wallet.

Everything seems to work fine when I use `arconnect` wallet but if I try with `arweave.app` wallet I get the next error message:

`deprecated, do not use`

I'm using the code provided from your docs here: https://academy.warp.cc/docs/sdk/advanced/plugins/deployment#signer

```js
const wallet = new ArweaveWebWallet({
  name: 'Your application name',
  logo: 'URL of your logo to be displayed to users',
});

wallet.setUrl('arweave.app');
await wallet.connect();
const userSigner = new InjectedArweaveSigner(wallet.namespaces.arweaveWallet);
await userSigner.setPublicKey();
const { contractTxId } = await warp.deploy({
  wallet: userSigner,
  initState: JSON.stringify(initialState),
  src: contractSrc,
});
```
The error `deprecated, do not use` bumps out after calling the `deploy` method.
```js
const { contractTxId } = await warp.deploy({
  wallet: userSigner,
  initState: JSON.stringify(initialState),
  src: contractSrc,
});
```

What do you think could be the problem? the error message `deprecated, do not use` seems very specific and it's weird that deployments work fine when using `arconnect`.

For `arconnect` I'm using this other code (provided in your docs):

```js
if (window.arweaveWallet) {
  await window.arweaveWallet.connect(['ACCESS_ADDRESS', 'SIGN_TRANSACTION', 'ACCESS_PUBLIC_KEY', 'SIGNATURE']);
}
const userSigner = new InjectedArweaveSigner(window.arweaveWallet);
await userSigner.setPublicKey();
const { contractTxId } = await warp.deploy({
  wallet: userSigner,
  initState: JSON.stringify(initialState),
  src: contractSrc,
});
```

Btw, if I disable bundling in the `deploy` method with the flag `disableBundling`, the deployment works fine with both wallets (arweave.app and arconnect). 

The problem seems to be related to the `InjectedArweaveSigner` object after calling the `setPublicKey` method when using `arweave.app` wallet.

Thanks in advance for your help, you rock!

**Comment by luckyr13** [comment]
Hello friends, it's me again. I found the problem.

The error message: `deprecated, do not use` comes from the `arweave-wallet-connector` when you call the `signature` method: https://github.com/jfbeats/ArweaveWalletConnector/blob/master/src/Arweave.ts#L77

So I think that the issue is related to this line in the `InjectedArweaveSigner` class:
https://github.com/warp-contracts/warp-contracts-plugins/blob/main/warp-contracts-plugin-deploy/src/signers/InjectedArweaveSigner.ts#L32

So, this is always gonna fail using the `arweave-wallet-connector`:
```js
await userSigner.setPublicKey();
```

I'm pretty sure that you are very busy, but if you can fix this it would be so awesome. 🙏

Thank you for your time and for helping us ❤️

**Comment by luckyr13** [comment]
Fixed in `warp-contracts` 1.4.1
